transport/rif_comms/client.py

```diff
@@ -8,95 +8,101 @@
         """
-        Subscribes to a pub-sub topic between self.rsk_address and partner_address.
-        Invokes CreateTopicWithRskAddress grpc api endpoint.
-        :param topic_id: ID of the topic to subscribe
-        :return: Notification stream
+        Subscribes to a pub-sub topic in order to send messages to or receive messages from an address.
+        Invokes CreateTopicWithRskAddress GRPC API endpoint.
+        The resulting notification stream should only be used for receiving messages; use send_message for sending.
+        :param rsk_address: destination RSK address for message sending
+        :return: notification stream for receiving messages
         """
-        # TODO catch already subscribed and any error
-        return self.stub.CreateTopicWithRskAddress(RskAddress(address=topic_id))
+        # TODO: catch already subscribed and any error
+        return self.stub.CreateTopicWithRskAddress(RskAddress(address=rsk_address))
 
-    # TODO review params and create docstring. It has no sense to use both peer id and rsk_address
-    def has_subscription(self, rsk_address: Address) -> BooleanResponse:
-        peer_id = self.get_peer_id(rsk_address)
+    def is_subscribed_to(self, rsk_address: Address) -> bool:
+        """
+        Returns whether or not the client's underlying RIF Communications node is subscribed to the topic
+        which corresponds to the given RSK address.
+        Invokes HasSubscriber GRPC API endpoint.
+        :param rsk_address: RSK address which corresponds to the topic which is being checked for subscription
+        :return: boolean value indicating whether the client is subscribed or not
+        """
+        our_peer_id = self._get_peer_id(self.rsk_address)
+        topic_id = self._get_peer_id(rsk_address)
         return self.stub.HasSubscriber(
             Subscriber(
-                peerId=peer_id,
-                channel=Channel(channelId=rsk_address)
+                peerId=our_peer_id,
+                channel=Channel(channelId=topic_id)
             )
-        )
+        ).value
 
-    def send_message(self, topic_id: str, data: str) -> Void:
+    def send_message(self, payload: str, rsk_address: Address):
         """
-        Sends a message to a topic.
-        Invokes the SendMessageToTopic grpc api endpoint
-        :param topic_id: topic identifier
-        :param message: the message data
-        :return: void
+        Sends a message to a destination RSK address.
+        Invokes the SendMessageToTopic GRPC API endpoint.
+        :param payload: the message data to be sent
+        :param rsk_address: the destination for the message to be sent to
         """
-
-        # TODO message encoding
-        return self.stub.SendMessageToTopic(
+        topic_id = self._get_peer_id(rsk_address)
+        # TODO: message encoding
+        self.stub.SendMessageToTopic(
             PublishPayload(
                 topic=Channel(channelId=topic_id),
```

Comment:
same here, im confussed about channelId must be an address or a peer id

i think that peer id is something that we must not know from lumino, but maybe the comms implementation need this

Reply:
btw, did you check that the test_rif_comms_client works with those modifications?

Reply:
> same here, im confussed about channelId must be an address or a peer id
> 
> i think that peer id is something that we must not know from lumino, but maybe the comms implementation need this

see [previous reply](https://github.com/rsksmart/lumino/pull/161#discussion_r527727683).

> btw, did you check that the test_rif_comms_client works with those modifications?

no, on it.

Reply:
@marcosmartinez7 is going to push the fixes based on the rif comms unit test results